### ISSUE TITLE
[angle] Build error

### ISSUE THREAD

**Issue by xiaozhuai** [microsoft/vcpkg#24396]
**Host Environment**
- MacOS 11.6.5 
- Compiler: Apple clang version 13.0.0 (clang-1300.0.29.3)

**To Reproduce**
`./vcpkg install angle:x64-osx-dynamic`

**Failure logs**
```
-- [COMMUNITY] Loading triplet configuration from: /Users/xiaozhuai/vcpkg/triplets/community/x64-osx-dynamic.cmake
-- Using cached google-angle-d15be77864e18f407c317be6f6bc06ee2b7d070a.tar.gz.
-- Cleaning sources at /Users/xiaozhuai/vcpkg/buildtrees/angle/src/ee2b7d070a-83a419ab43.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/xiaozhuai/vcpkg/downloads/google-angle-d15be77864e18f407c317be6f6bc06ee2b7d070a.tar.gz
-- Applying patch 001-fix-uwp.patch
-- Applying patch 002-fix-builder-error.patch
-- Applying patch 003-fix-mingw.patch
-- Using source at /Users/xiaozhuai/vcpkg/buildtrees/angle/src/ee2b7d070a-83a419ab43.clean
-- Using cached /Users/xiaozhuai/vcpkg/downloads/angle-09490503d0f201b81e03f5ca0ab8ba8ee76d4a8e.tar.gz
-- Extracting source /Users/xiaozhuai/vcpkg/downloads/angle-09490503d0f201b81e03f5ca0ab8ba8ee76d4a8e.tar.gz
-- Applying patch third-party-zlib-far-undef.patch
-- Using source at /Users/xiaozhuai/vcpkg/buildtrees/angle/src/8ee76d4a8e-2dd6a50ec1.clean
-- Configuring x64-osx-dynamic-dbg
-- Configuring x64-osx-dynamic-rel
-- Building x64-osx-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: /usr/local/Cellar/cmake/3.23.0/bin/cmake --build . --config Debug --target install -- -v -j21
    Working Directory: /Users/xiaozhuai/vcpkg/buildtrees/angle/x64-osx-dynamic-dbg
    See logs for more information:
      /Users/xiaozhuai/vcpkg/buildtrees/angle/install-x64-osx-dynamic-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:106 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/angle/portfile.cmake:64 (vcpkg_install_cmake)
  scripts/ports.cmake:145 (include)


error: building angle:x64-osx-dynamic failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+angle
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[angle]+Build+error
including:
  package: angle[core]:x64-osx-dynamic -> chromium_4472#2
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: 2ac61f87f 2022-04-22 (3 days ago)
```

[install-x64-osx-dynamic-dbg-out.log](https://github.com/microsoft/vcpkg/files/8552567/install-x64-osx-dynamic-dbg-out.log)

**Comment by JackBoosY** [comment]
```
[339/341] : && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -fPIC -g -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -mmacosx-version-min=11.6 -dynamiclib -Wl,-headerpad_max_install_names -framework Cocoa -framework OpenGL -framework IOKit -framework CoreFoundation -framework IOSurface -framework QuartzCore -framework CoreGraphics -o libGLESv2.dylib -install_name @rpath/libGLESv2.dylib CMakeFiles/angle_common.dir/src/common/Float16ToFloat32.cpp.o CMakeFiles/angle_common.dir/src/common/MemoryBuffer.cpp.o CMakeFiles/angle_common.dir/src/common/PackedEGLEnums_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/PackedEnums.cpp.o CMakeFiles/angle_common.dir/src/common/PackedGLEnums_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/PoolAlloc.cpp.o CMakeFiles/angle_common.dir/src/common/aligned_memory.cpp.o CMakeFiles/angle_common.dir/src/common/android_util.cpp.o CMakeFiles/angle_common.dir/src/common/angleutils.cpp.o CMakeFiles/angle_common.dir/src/common/debug.cpp.o CMakeFiles/angle_common.dir/src/common/entry_points_enum_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/mathutil.cpp.o CMakeFiles/angle_common.dir/src/common/matrix_utils.cpp.o CMakeFiles/angle_common.dir/src/common/string_utils.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils_apple.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils_mac.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils_posix.cpp.o CMakeFiles/angle_common.dir/src/common/third_party/base/anglebase/sha1.cc.o CMakeFiles/angle_common.dir/src/common/third_party/smhasher/src/PMurHash.cpp.o CMakeFiles/angle_common.dir/src/common/third_party/xxhash/xxhash.c.o CMakeFiles/angle_common.dir/src/common/tls.cpp.o CMakeFiles/angle_common.dir/src/common/uniform_type_info_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/utilities.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/egl_ext_stubs.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/egl_stubs.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_egl_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_egl_ext_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_1_0_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_2_0_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_3_0_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_3_1_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_3_2_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_ext_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/global_state.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/libGLESv2_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/proc_table_egl_autogen.cpp.o  -Wl,-rpath,/Users/xiaozhuai/vcpkg/installed/x64-osx-dynamic/debug/lib  libANGLE.a  /Users/xiaozhuai/vcpkg/installed/x64-osx-dynamic/debug/lib/libz.dylib  -framework  IOKit  -framework  CoreFoundation  -framework  CoreGraphics && :
FAILED: libGLESv2.dylib 
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -fPIC -g -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -mmacosx-version-min=11.6 -dynamiclib -Wl,-headerpad_max_install_names -framework Cocoa -framework OpenGL -framework IOKit -framework CoreFoundation -framework IOSurface -framework QuartzCore -framework CoreGraphics -o libGLESv2.dylib -install_name @rpath/libGLESv2.dylib CMakeFiles/angle_common.dir/src/common/Float16ToFloat32.cpp.o CMakeFiles/angle_common.dir/src/common/MemoryBuffer.cpp.o CMakeFiles/angle_common.dir/src/common/PackedEGLEnums_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/PackedEnums.cpp.o CMakeFiles/angle_common.dir/src/common/PackedGLEnums_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/PoolAlloc.cpp.o CMakeFiles/angle_common.dir/src/common/aligned_memory.cpp.o CMakeFiles/angle_common.dir/src/common/android_util.cpp.o CMakeFiles/angle_common.dir/src/common/angleutils.cpp.o CMakeFiles/angle_common.dir/src/common/debug.cpp.o CMakeFiles/angle_common.dir/src/common/entry_points_enum_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/mathutil.cpp.o CMakeFiles/angle_common.dir/src/common/matrix_utils.cpp.o CMakeFiles/angle_common.dir/src/common/string_utils.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils_apple.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils_mac.cpp.o CMakeFiles/angle_common.dir/src/common/system_utils_posix.cpp.o CMakeFiles/angle_common.dir/src/common/third_party/base/anglebase/sha1.cc.o CMakeFiles/angle_common.dir/src/common/third_party/smhasher/src/PMurHash.cpp.o CMakeFiles/angle_common.dir/src/common/third_party/xxhash/xxhash.c.o CMakeFiles/angle_common.dir/src/common/tls.cpp.o CMakeFiles/angle_common.dir/src/common/uniform_type_info_autogen.cpp.o CMakeFiles/angle_common.dir/src/common/utilities.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/egl_ext_stubs.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/egl_stubs.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_egl_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_egl_ext_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_1_0_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_2_0_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_3_0_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_3_1_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_3_2_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/entry_points_gles_ext_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/global_state.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/libGLESv2_autogen.cpp.o CMakeFiles/libGLESv2.dir/src/libGLESv2/proc_table_egl_autogen.cpp.o  -Wl,-rpath,/Users/xiaozhuai/vcpkg/installed/x64-osx-dynamic/debug/lib  libANGLE.a  /Users/xiaozhuai/vcpkg/installed/x64-osx-dynamic/debug/lib/libz.dylib  -framework  IOKit  -framework  CoreFoundation  -framework  CoreGraphics && :
Undefined symbols for architecture x86_64:
  "rx::CreateDisplayCGLOrEAGL(egl::DisplayState const&)", referenced from:
      egl::(anonymous namespace)::CreateDisplayFromAttribs(long, long, long, egl::DisplayState const&) in libANGLE.a(Display.cpp.o)
ld: symbol(s) not found for architecture x86_64
```

**Comment by xiaozhuai** [comment]
Fixed by #24750
